### PR TITLE
Fix for https://github.com/nickg/nvc/issues/1046 on macos

### PR DESCRIPTION
**Pull request by jeisch**
Adding LDFLAGS for zstd from pkg-config to resolve issue https://github.com/nickg/nvc/issues/1046

### CODE REVIEW

File: m4/ax_llvm_c.m4 [hunk 1]
```diff
@@ -118,6 +118,15 @@ AC_DEFUN([AX_LLVM_C], [
                          [LLVM uwtable attribute takes an argument])
     fi
 
+    if test "$llvm_ver_num" -ge "160"; then
+      case $host_os in
+      darwin*)
+        # Add LDFLAGS for libzstd in a non-default location for issue #1046
+        ZSTD_LDFLAGS=`pkg-config --libs-only-L libzstd`
```
Comment (nickg) [inline m4/ax_llvm_c.m4:125]: It's probably better to use the `PKG_CHECK_MODULES` macro here. Does this work?

```
PKG_CHECK_MODULES([libzstd], [libzstd >= 1.4])
LLVM_LDFLAGS="$LLVM_LDFLAGS $libzstd_LDFLAGS"
```

Comment (nickg) [reply m4/ax_llvm_c.m4:125]: Probably should be `LLVM_LIBS` instead of `LLVM_LDFLAGS`.

**Comment by jeisch** [comment]
I have incorporated the comments into my most recent commit to the branch and tested successfully on my machine.

**Comment by nickg** [comment]
I moved this around a bit in 41d7f9f2f40f to avoid a warning about duplicated `-lstd`. Can you check it still works for you?

**Comment by jeisch** [comment]
On my system, `-lzstd` is provided by both `pkg-config --libs libzstd` (the source of $libzstd_LIBS) and by `llvm-config --system-libs` so it is still duplicated after the changes in https://github.com/nickg/nvc/commit/41d7f9f2f40ff48307a2fbdbb464cdcaa41d6c17 

>  % pkg-config --libs libzstd
-L/opt/local/lib -lzstd
 % llvm-config --system-libs
-lm -lz -lzstd -lcurses -lxml2

**Comment by nickg** [comment]
OK I think I misunderstood the problem a bit and missed that you have `--libs-only-L` in the original patch. Can you test #1056? It's basically the same as your original patch but without hard-coding `pkg-config`.

**Pull request merged**